src/server/protocol.ts

```diff
@@ -630,6 +631,26 @@ export interface GetMoveToRefactoringFileSuggestions extends Response {
     };
 }
 
+/**
+ * Request refactorings at a given position post pasting text from some other location.
+ */
+
+export interface GetPostPasteImportFixesRequest extends Request {
+    command: CommandTypes.GetPostPasteImportFixes;
+    arguments: GetPostPasteImportFixesRequestArgs;
+}
+
+export type GetPostPasteImportFixesRequestArgs = FileRequestArgs & {
+    pastes: { text: string; range: TextSpan; }[];
+    copySpan?: FileSpan;

```

Comment:
For the first version of copy/paste the idea was to paste the same text at multiple cursors. So yes, with the current protocol if a single copy had to be pasted at 100 cursors, the text would be duplicated 100 times in `pastes`. Whenever later we decided to allow copying different text at multiple cursors and pasting at multiple cursor locations, the protocol would have to change to `copySpan?: FileSpan[]`.
Looking at your mental model, I think that seems to make this clearer 😄 I'd be happy to change this part of the protocol if that's more useful!

Reply:
The protocol can’t really change; we can only add to it. So even if we ship an implementation that doesn’t cover all the use cases we’d eventually like to support, the protocol design needs to be somewhat future proof. (That said, from discussions so far, it didn’t sound like it will be very hard to implement a paste provider that’s faithful to the way VS Code multi-cursor paste usually works. Since this isn’t supposed to land until 5.5, I think there’s likely plenty of time to make that happen.)